crates/next-core/src/next_route_matcher/prefix_suffix.rs

```diff
@@ -0,0 +1,65 @@
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use turbo_tasks::primitives::BoolVc;
+use turbopack_node::route_matcher::{ParamsVc, RouteMatcher};
+
+/// A composite route matcher that matches a path if it has a given prefix and
+/// suffix.
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct PrefixSuffixMatcher<T>
+where
+    T: RouteMatcher,
+{
+    prefix: String,
+    suffix: String,
+    inner: T,
+}
+
+impl<T> PrefixSuffixMatcher<T>
+where
+    T: RouteMatcher,
+{
+    /// Creates a new [PrefixSuffixMatcher].
+    pub fn new(prefix: String, suffix: String, inner: T) -> Self {
+        Self {
+            prefix,
+            suffix,
+            inner,
+        }
+    }
+
+    fn strip_prefix_and_suffix<'a, 'b>(&'a self, path: &'b str) -> Result<Option<&'b str>> {
```

Comment:
Nit: Can we unnest this?

```suggestion
    fn strip_prefix_and_suffix<'a, 'b>(&'a self, path: &'b str) -> Option<&'b str> {